download_ssara_rsmas.py

```diff
@@ -7,149 +7,150 @@
+            logger.log(logging.WARNING, "The file, %s, didn't download correctly. Running ssara again.")
+            run_ssara(run_number + 1)
+            return
+
+    logger.log(logging.INFO, "Everything is there!")
 
 
 def run_ssara(run_number=1):
-	""" Runs ssara_federated_query-cj.py and checks for download issues.
-	
-		Runs ssara_federated_query-cj.py and checks continuously for whether the data download has hung without 
-		comleting or exited with an error code. If either of the above occur, the function is run again, for a 
-		maxiumum of 10 times.
-		
-		Parameters: run_number: int, the current iteration the wrapper is on (maxiumum 10 before quitting)
-		Returns: status_cod: int, the status of the donwload (0 for failed, 1 for success)
-
-	"""	
-
-
-	logger.info("RUN NUMBER: %s", str(run_number))	
-	if run_number > 10:
-		return 0	
-	
-	# Compute SSARA options to use 
-	with open(inps.template, 'r') as template_file:
-		options = ''
-		for line in template_file:
-			if 'ssaraopt' in line:
-				options = line.strip('\n').rstrip().split("= ")[1]
-				break;
-	options = options.split(' ')
-
-	# Runs ssara_federated_query-cj.py with proper options
-	ssara_options = ['ssara_federated_query-cj.py'] + options + ['--parallel', '10', '--print', '--download']	
-	ssara_process = subprocess.Popen(ssara_options)
-		
-	completion_status = ssara_process.poll()	# the completion status of the process
-	hang_status = False							# whether ot not the download has hung
-	wait_time = 10								# wait time in 'minutes' to determine hang status
-	prev_size = -1								# initial download directory size
-	i=0											# the iteration number (for logging only)
-	
-	# while the process has not completed
-	while completion_status is None:
-		
-		i=i+1
-		
-		# Computer the current download directory size
-		curr_size = int(subprocess.check_output(['du','-s', os.getcwd()]).split()[0].decode('utf-8'))
-
-		# Compare the current and previous directory sizes to determine determine hang status
-		if prev_size == curr_size:
-			hang_status = True
-			logger.warning("SSARA Hung")
-			ssara_process.terminate()	# teminate the process beacause download hung
-			break;						# break the completion loop 
-		
-		time.sleep(60*wait_time)		# wait 'wait_time' minutes before continuing
-		prev_size = curr_size
-		completion_status = ssara_process.poll()
-		logger.info("{} minutes: {:.1f}GB, completion_status {}".format(i*wait_time, curr_size/1024/1024, completion_status))
-			
-	exit_code = completion_status 	# get the exit code of the command
-	logger.info("EXIT CODE: %s", str(exit_code))
-	
-	bad_codes = [137]
-	
-	# If the exit code is one that signifies an error, rerun the entire command
-	if exit_code in bad_codes or hang_status:
-		logger.warning("Something went wrong, running again")
-		run_ssara(run_number=run_number+1)
-
-	return 1
+    """ Runs ssara_federated_query-cj.py and checks for download issues.
```

Comment:
Could you remove unchanged lines from this diff?

Reply:
It's a bit hard to read and takes away from the important parts of the diff

Reply:
How do I do this?

Reply:
Also, this makes `git blame` more useful too